packages/page-bounties/src/BountyCreate.tsx

```diff
@@ -6,24 +6,38 @@ import type { BalanceOf } from '@polkadot/types/interfaces';
 import BN from 'bn.js';
 import React, { useCallback, useState } from 'react';
 
+import { DeriveBalancesAll } from '@polkadot/api-derive/types';
 import { Button, Input, InputAddress, InputBalance, Modal, TxButton } from '@polkadot/react-components';
-import { useApi, useToggle } from '@polkadot/react-hooks';
+import { useApi, useCall, useToggle } from '@polkadot/react-hooks';
 import { BN_ZERO } from '@polkadot/util';
 
-import { calculateBountyBond } from './helpers/calculateBountyBond';
+import { calculateBountyBond, countUtf8Bytes } from './helpers/calculateBountyBond';
 import { useTranslation } from './translate';
 
+const MIN_TITLE_LEN = 1;
+const TITLE_DEFAULT_VALUE = '';
+const BOUNTY_DEFAULT_VALUE = BN_ZERO;
+
 function BountyCreate () {
   const { t } = useTranslation();
   const { api } = useApi();
 
   const [accountId, setAccountId] = useState<string | null>(null);
   const [title, setTitle] = useState('');
   const [bond, setBond] = useState(((api.consts.bounties || api.consts.treasury).bountyDepositBase as BalanceOf).toBn());
-  const [amount, setAmount] = useState<BN | undefined>(BN_ZERO);
-  const [isValid] = useState(true);
+  const [value, setValue] = useState<BN | undefined>(BOUNTY_DEFAULT_VALUE);
   const [isOpen, toggleIsOpen] = useToggle();
 
+  const balances = useCall<DeriveBalancesAll>(api.derive.balances.all, [accountId]);
+  const bountyMinValue = ((api.consts.bounties || api.consts.treasury).bountyValueMinimum as BalanceOf).toBn();
+  const bountyTitleMaxLength = ((api.consts.bounties || api.consts.treasury).maximumReasonLength as BalanceOf).toNumber();
+
+  const isTitleValid = title?.length >= MIN_TITLE_LEN && countUtf8Bytes(title) <= bountyTitleMaxLength;
+  const isValueValid = value?.gte(bountyMinValue);
+  const hasFunds = balances?.availableBalance.gte(bond);
```

Comment:
Not convinced about all these re-calculated each time a character is typed, e.g. on render. Probably would have opted for a useEffect (maybe even with a debounce via `react-hooks/useDebounce`)